app.go

```diff
@@ -191,6 +191,24 @@ func (a *App) useShortOptionHandling() bool {
 func (a *App) Run(arguments []string) (err error) {
 	a.Setup()
 
+	commandNamesWithoutDuplicates := make(map[string]bool)
+	for _, c := range a.Commands {
+		_, cExists := commandNamesWithoutDuplicates[c.Name]
+		if cExists {
+			return fmt.Errorf("list of commands contains duplicate Name: %s", c.Name)
+		}
+		commandNamesWithoutDuplicates[c.Name] = true
+
+		subCommandNamesWithoutDuplicates := make(map[string]bool)
+		for _, s := range c.Subcommands {
+			_, sExists := subCommandNamesWithoutDuplicates[s.Name]
+			if sExists {
+				return fmt.Errorf("list of child commands of %s contains duplicate Name: %s", c.Name, s.Name)
+			}
+			subCommandNamesWithoutDuplicates[s.Name] = true
+		}
```

Comment:
⭐⭐⭐ _(must change)_ subcommands can have other subcommands within them, so you'll want to write this as a recursion or similar.

Reply:
Struggling a little trying to figure this out. Give me a little more time. Thanks

Reply:
I have created the recursive function in this commit.
https://github.com/urfave/cli/pull/984/commits/021127a013aaf344c97f0efcae38904e4dff1fa3
It's working so far but I'm yet to refine it and add more test cases. Will update you guys once it's ready :)